### PR TITLE
[WIP] Add mongo backend to tsung test framework

### PR DESCRIPTION
**Pull request by NoahCarnahan**
@snopoke @czue cc @dannyroberts 
Getting very close to being able to run the mongo benchmarks, I hope.

### CODE REVIEW

File: tsung/backends.py [hunk 1]
```diff
@@ -215,3 +216,33 @@ def load_data(self, dest_folder):
...
+        self.mongo = get_mongo(self.name)
+
+    def _create_user(self):
+        return User(id=str(uuid4()), username='admin', password='secret')
+
+    def reset_db(self):
+        super(PrototypeMongo, self).reset_db()
+
+        print('Dropping mongo')
+        self.mongo(eval="'db.dropDatabase()'")
```
Comment (czue) [inline tsung/backends.py]: lol, this is beautiful and so wrong

Comment (snopoke) [inline tsung/backends.py]: Wow, crazy

**Comment by snopoke** [comment]
Looking good. Assuming you'll self merge when ready.

File: prototype/mobile_endpoint/views/receiver.py [hunk 2]
```diff
@@ -25,6 +26,7 @@ def couch_receiver(domain):
 @ota_mod.route('/mongo-receiver/<domain>', methods=['POST'])
 @requires_auth
 def mongo_receiver(domain):
+    connect(host=current_app.config.get('MONGO_URI'))
```
Comment (NoahCarnahan) [inline prototype/mobile_endpoint/views/receiver.py:29]: Initiating the `mongoengine` connection here feels wrong, but didn't spend the time to figure out the "right" place to put this. (Docs seem to suggest at the top of the models file, but `flask.current_app` isn't always known when that is first imported...)

Comment (snopoke) [inline prototype/mobile_endpoint/views/receiver.py:29]: Can you not do it in `mobile_endpoint.__init__.create_app` ?

On 6 August 2015 at 21:30, Noah Carnahan notifications@github.com wrote:

> In prototype/mobile_endpoint/views/receiver.py
> https://github.com/dimagi/mobile-endpoint/pull/45#discussion_r36455430:
> 
> > @@ -25,6 +26,7 @@ def couch_receiver(domain):
> >  @ota_mod.route('/mongo-receiver/<domain>', methods=['POST'])
> >  @requires_auth
> >  def mongo_receiver(domain):
> > -    connect(host=current_app.config.get('MONGO_URI'))
> 
> Initiating the mongoengine connection here feels wrong, but didn't spend
> the time to figure out the "right" place to put this. (Docs seem to suggest
> at the top of the models file, but flask.current_app isn't always known
> when that is first imported...)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/dimagi/mobile-endpoint/pull/45/files#r36455430.

## 

Simon Kelly
Senior Engineer | Dimagi South Africa

**Pull request merged**